components/sup/src/manager/service/health.rs

```diff
@@ -1,19 +1,24 @@
-// Copyright (c) 2017 Chef Software Inc. and/or applicable contributors
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-use std::fmt;
+use crate::manager::{service::{hook_runner,
+                               hooks::HealthCheckHook,
+                               supervisor::Supervisor},
+                     GatewayState};
+use futures::{future::{self,
+                       lazy,
+                       Either,
+                       Future,
+                       Loop},
+              IntoFuture};
+use habitat_common::templating::package::Pkg;
+use habitat_core::service::{HealthCheckInterval,
+                            ServiceGroup};
+use std::{fmt,
+          ops::Deref,
+          sync::{Arc,
+                 Mutex,
+                 RwLock},
+          time::Instant};
 
+/// The possible results of running a health check hook.
```

Comment:
Maybe this implies that `HealthCheck` isn't really the best name. Perhaps `ServiceHealth` or `HealthCheckResponse` would be better.

Reply:
Indeed... I'll see what the implications of such a rename would be.

Reply:
All good!